astropy/visualization/interval.py

```diff
@@ -283,9 +283,9 @@ def get_limits(self, values):
             last_ngoodpix = ngoodpix
             ngoodpix = np.sum(~badpix)
 
-        slope, intercept = fit
-
         if ngoodpix >= minpix:
```

Comment:
Do we need a second condition here?
```suggestion
        if ngoodpix >= minpix and ngoodpix < last_ngoodpix:
```

Reply:
No, this isn't needed.  For the first iteration `last_ngoodpix = ngoodpix + 1`, thus it will never `break` without running the `fit` at least once.